pyiron_base/generic/flattenedstorage.py

```diff
@@ -796,6 +798,60 @@ def add_chunk(self, chunk_length, identifier=None, **arrays):
         self.current_element_index = i
         # return last_chunk_index, last_element_index
 
+    def extend_storage(self, other: "FlattenedStorage"):
+        self._check_compatible_fill_values(other=other)
+
+        combined_num_chunks = self.num_chunks + other.num_chunks
+        combined_num_elements = self.num_elements + other.num_elements
+        if combined_num_chunks > self._num_chunks_alloc:
+            self._resize_chunks(combined_num_chunks)
+        if combined_num_elements > self._num_elements_alloc:
+            self._resize_elements(combined_num_elements)
+
+        for k, a in other._per_chunk_arrays.items():
+            # add start_index of last chunk to start_index of other for correct mapping
+            if k == "start_index":
+                a += self._per_chunk_arrays[k][self.num_chunks]
```

Comment:
I think this modifies the `other` storage in place.  Double check that in tests.